rust/src/combinators/chain/ffi.rs

```diff
@@ -7,7 +7,8 @@ use crate::ffi::any::{AnyMeasurement, AnyTransformation};
 
 #[bootstrap(
     features("contrib"),
-    arguments(measurement1(rust_type = b"null"), transformation0(rust_type = b"null"))
+    arguments(measurement1(rust_type = b"null"), transformation0(rust_type = b"null")),
+    dependencies("$get_dependencies(measurement1)", "$get_dependencies(transformation0)")
```

Comment:
I'm not convinced this will always do the right thing. In most of the combinators, the arguments get cloned. So the dependency here would be on the original argument, not what actually ends up being used. It works (I think) because internally, the clone is implemented by bumping the count on the `Rc`, so it's the same object. Seems like a better approach would be for Rust to declare the dependencies (though not sure how to do that).

The Apache Arrow memory model may be instructive here.

Reply:
The dependencies in `measurement1` are the c wrappers for the function and privacy map. So the Rc we're keeping from going to zero is not a Rust Rc, it's a Python one. This approach is to appease the python refcounter only.

Reply:
After our discussion, I see how this is OK: Cloning could only ever reduce the number of dependencies, so this should always be safe.